pkg/scheduler/core/generic_scheduler.go

```diff
@@ -648,6 +648,12 @@ func PrioritizeNodes(
 			}(i, priorityConfig)
 		} else {
 			results[i] = make(schedulerapi.HostPriorityList, len(nodes))
+			for index, node := range nodes {
```

Comment:
How is this going to fix issue #68561? The issue indicates that if Map function fails for index "i", `results[i+1]` (and later) will not be be populated. So, it looks to me that the [this `return`](https://github.com/kubernetes/kubernetes/blob/5e5f7e53896a171959c34483a60db5770a8abac0/pkg/scheduler/core/generic_scheduler.go#L663) statement should be removed.

Reply:
I think #68561 described that when Map function fails for `i`, it panics when calling Reduce function of `i`.

In current logic, if Map function fails, HostPriority at index `i` holds "default" value: i.e. {Host: "", Score: 0}. Then after Map, it tries to call specific ReduceFunctions:

https://github.com/kubernetes/kubernetes/blob/5e5f7e53896a171959c34483a60db5770a8abac0/pkg/scheduler/core/generic_scheduler.go#L675

Take `selector_spreading` priority for example, it panics at:

https://github.com/kubernetes/kubernetes/blob/5e5f7e53896a171959c34483a60db5770a8abac0/pkg/scheduler/algorithm/priorities/selector_spreading.go#L126

because `result[i].Host` is "", then `nodeNameToInfo[""].Node()` would panic.

Based on above assumption, I'm fine with this fix. But an alternative fix is to simply bypass the Reduce function if corresponding Map function failed.

Reply:
> I think #68561 described that when Map function fails for i, it panics when calling Reduce function of i.

That's not exactly what I understand from [the following statement](https://github.com/kubernetes/kubernetes/issues/68561#issuecomment-420668834):

> As shown above, if priorityConfigs[i].Map(pod, meta, nodeInfo) returns with error, there will be no chance for results[i + 1] to initialize its value, thus causing panic when calling
config.Reduce(pod, meta, nodeNameToInfo, results[i+1])

Reply:
Anyway, my comment about remove the `return` statement is incorrect. I missed the fact that the return statement is in a function closure.

Reply:
@DylanBLE could you clarify a little bit? Is it (1) failure of map function on `i` panics reduce function `i`, or (2) failure of map function on `i` panics reduce function `i+1`

Reply:
@bsalamat My understanding was on the same lines as @Huang-Wei. We should be good with this fix in terms of correctness but in terms of performance we can have what you suggested with improvement suggested by @Huang-Wei.